### DISCUSSION TITLE
Problem Using Cyfrin DevopsTool

### DISCUSSION THREAD

**Discussion by SpringxDSay** [Cyfrin/foundry-full-course-cu#2322] [Q&A] [answered]
I installed the devops tools and updated my foundry.toml file as instructed. But I keep getting the error "get_most_recent_deployment not found".  While compiling my code, this is the error message I got: 
```
Error: 
Compiler run failed:
Error (9582): Member "get_most_recent_deployment" not found or not visible after argument-dependent lookup in type(library DevOpsTools).
  --> script/Interactions.s.sol:85:38:
   |
85 |         address mostRentlyDeployed = DevOpsTools.get_most_recent_deployment(Raffle, block.chainid);
```
I followed all the steps on installation in the Cyfrin/DevOps repo. So, what could be the issue?

**Comment by EngrPips** [comment]
Hello @SpringxDSay, Can you open the `DevOpsTools` file and search if there is a `get_most_recent_deployment` function in it?

**Reply by SpringxDSay** [reply]
I checked already. This is from the DevOpsTools.sol file:
```
 function get_most_recent_deployment(
        string memory contractName,
        uint256 chainId,
        string memory relativeBroadcastPath
    ) internal view returns (address) {}
    ```

**Reply by SpringxDSay** [reply, marked as answer]
So apparently, I was passing the the raffle contract as contract instead of string. So the correct thing is:
```
 address mostRentlyDeployed = DevOpsTools.get_most_recent_deployment("Raffle", block.chainid);
 ```

**Comment by EngrPips** [comment]
Hello @SpringxDSay, You were calling the wrong function because of the type of argument you passed in, the `get_most_recent_deployment` function expected an argument of type string and uint256 but you passed in argument of type `Raffle` and uint256 as shown in the below code which is yours 

```js
address mostRentlyDeployed = DevOpsTools.get_most_recent_deployment(Raffle, block.chainid);
```

Please do the below instead and let see if it works 

```js
address mostRentlyDeployed = DevOpsTools.get_most_recent_deployment("Raffle", block.chainid);
```

**Reply by SpringxDSay** [reply]
Yeah. I figured it out. Ty ser. Drop your x handle for a follow please

**Reply by EngrPips** [reply]
I am @engrpips on twitter.
